salt/client/mixins.py

```diff
@@ -433,7 +433,7 @@ def print_async_event(self, suffix, event):
         # more general, since this will get *really* messy as
         # people use more events that don't quite fit into this mold
         if suffix == 'ret':  # for "ret" just print out return
-            salt.output.display_output(event['return'], None, self.opts)
+            salt.output.display_output(event['return'], self.opts.get('out', None), self.opts)
```

Comment:
If an output option is not explicitely specified on the command-line, the behaviour is exactly the same as it was before.